params.go

```diff
@@ -759,6 +760,9 @@ func (p *Params) ValidateRequiredProperties() (bool, []error, []string) {
 		if p.Visibility == "esp" && (p.DisableServiceAccountKeyRotation == nil || !*p.DisableServiceAccountKeyRotation) {
 			errors = append(errors, fmt.Errorf("With visibility 'esp' property disableServiceAccountKeyRotation is required; set disableServiceAccountKeyRotation: true on this stage"))
 		}
+		if p.Visibility == "esp" && (p.EspEnpointsProjectID == nil) {
```

Comment:
`EspEnpointsProjectID` is a string, you can't compare it with `nil`, you should compare it with empty string (`""`)